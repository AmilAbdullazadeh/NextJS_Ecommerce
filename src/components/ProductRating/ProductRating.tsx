import React from 'react';
import Box from '@material-ui/core/Box';
import Rating from '@material-ui/lab/Rating';

import { Typography } from '@components';

export interface ProductRatingProps {
    ratingValue: number;
    reviewCount: number;
}

const ProductRating: React.FC<ProductRatingProps> = ({
    ratingValue,
    reviewCount,
}: ProductRatingProps) => (
    <Box display="flex">
        <Box>
            <Rating
                name="rating"
                value={ratingValue}
                readOnly
                precision={0.5}
                size="small"
            />
        </Box>
        <Box>
            <Typography variant="body2" color="primary">
                {reviewCount}
            </Typography>
        </Box>
    </Box>
);

export default ProductRating;
