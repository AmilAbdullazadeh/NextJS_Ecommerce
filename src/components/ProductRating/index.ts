export { default as ProductRating } from './ProductRating';
