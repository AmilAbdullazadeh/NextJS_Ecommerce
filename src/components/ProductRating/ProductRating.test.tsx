import React from 'react';

import { render } from '@testUtils';

import ProductRating, { ProductRatingProps } from './ProductRating';

describe('ProductRating', () => {
    const defaultProps: ProductRatingProps = {
        ratingValue: 3.5,
        reviewCount: 10,
    };

    it('matches snapshot', () => {
        const { container } = render(<ProductRating {...defaultProps} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
