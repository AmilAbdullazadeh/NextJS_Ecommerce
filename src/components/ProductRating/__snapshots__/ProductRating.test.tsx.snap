// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRating matches snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-1"
>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <span
      aria-label="3.5 Stars"
      class="MuiRating-root MuiRating-sizeSmall MuiRating-readOnly"
      role="img"
    >
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 50%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorPrimary"
    >
      10
    </p>
  </div>
</div>
`;
